src/clover/fileparser.py

```diff
@@ -2177,6 +2189,62 @@ def _parse_transmission_inputs(
     )
 
 
+def _parse_grid_inputs(
+    inputs_directory_relative_path: str,
+    logger: Logger,
+) -> List[GRID]:
+    """
+    Parses the grid inputs file.
+
+    Inputs:
+        - inputs_directory_relative_path:
+            The relative path to the inputs folder directory.
+        - logger:
+            The :class:`logging.Logger` to use for the run.
+        - scenarios:
+            The list of scenarios.
+
+    Outputs:
+        - The grid types (CURRENT.DRAW AND DAILY POWER).
+        - The grid tiers for each types.
+        - The grid costs for each tiers.
+        - The overall grid emissions (similar for all types and tiers).
+
+    """
+    # Parse the grid input information.
+    grid_inputs_filepath = os.path.join(
+        inputs_directory_relative_path, GRID_INPUTS_FILE
+    )
+    grid_inputs = read_yaml(
+        grid_inputs_filepath,
+        logger,
+    )
+    if not isinstance(grid_inputs, dict):
+        raise InputFileError("Grid inputs", "Grid input file is not of type `list`.")
+    logger.info("Grid inputs successfully parsed.")
+
+    grids: List[Grid] = []
+    for entry in grid_inputs["grids"]:
+        type: List[GridType]= []
+        tiers: List[GridTier] = []
+        for tier_entry in entry["tiers"]:
+                tiers.append(
+                    GridTier(
+                        tier_entry["upper_bound"]["consumption"],
+                        tier_entry["costs"],
+                    )
+                )
+        type.append(
+            GridType(
+                entry["type"],
+            )
+        )
+        print(type)
+        grids.append(Grid(entry["name"],type, tiers))
+    # grid_emissions = grid_inputs["emissions"]  # the same for all the grids (EDL,Diesel)
+    return (grids) #grid_emissions can be added here
+
```

Comment:
Kindly check this section, I changed the grid_input.yaml for it to make more sense and changed this accordingly. @BenWinchester